### PR TITLE
feat: enhancement of dark mode

### PR DESCRIPTION
**Pull request by CaCaBlocker**
Because

- When re-render the page or click the exterior link in the content. The dark mode state will be wiped up.

This commit

- Create a custom hook to store and extract the dark mode state on localStorage.

### CODE REVIEW

File: src/components/docs/ThemeToggle/ThemeToggle.tsx [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
+import { useCallback, useEffect, useState } from "react";
 import cn from "clsx";
 
 import { MoonIcon } from "./MoonIcon";
 import { SunIcon } from "./SunIcon";
 
-export const ThemeToggle = () => {
-  const [isDark, setIsDark] = useState(false);
+//  Custom Hook for dark mode
+export const useDark = () => {
```
Comment (EiffelFly) [inline src/components/docs/ThemeToggle/ThemeToggle.tsx:8]: I would suggest that separate this hook into dedicated file and put it in the `/hooks` folder, the name can be `useDarkTheme`. And please rename the `[value, setValue]` to something like `[isDark, setIsDark]` We should make the naming as explicit as possible

**Review by EiffelFly — COMMENTED** [review]
@CaCaBlocker Generally, I think this code is clean and you had already added the comment to describe the logic, nice! But I would suggest that we don't put the hook inside a component, please follow the suggestion.

Thank for your contribution!!

File: src/components/docs/ThemeToggle/ThemeToggle.tsx (hunk 1 above)
Comment (CaCaBlocker) [reply src/components/docs/ThemeToggle/ThemeToggle.tsx:8]: Okay, I will follow that.

Comment (CaCaBlocker) [reply src/components/docs/ThemeToggle/ThemeToggle.tsx]: @EiffelFly I just updated the code. Please check that.

File: src/components/docs/ThemeToggle/ThemeToggle.tsx [hunk 2]
```diff
@@ -1,22 +1,16 @@
...
+import useDarkTheme from "@/hooks/useDarkTheme";
+
 import { MoonIcon } from "./MoonIcon";
 import { SunIcon } from "./SunIcon";
 
 export const ThemeToggle = () => {
-  const [isDark, setIsDark] = useState(false);
+  const [isDark, setIsDark] = useDarkTheme();
 
   const toggleTheme = useCallback(() => {
```
Comment (EiffelFly) [inline src/components/docs/ThemeToggle/ThemeToggle.tsx:12]:
```suggestion
  const toggleTheme = useCallback(() => {
    setIsDark(state => !state)
  }, []]
```

@CaCaBlocker You can do this to remove the deps

Comment (CaCaBlocker) [reply src/components/docs/ThemeToggle/ThemeToggle.tsx:12]: Yeah, you're right. I just updated it.

Comment (CaCaBlocker) [reply src/components/docs/ThemeToggle/ThemeToggle.tsx:12]: @EiffelFly I just added deps. Till now, I tried to use npm(install & lint), not pnpm. Certainly, this project is working well with pnpm.

**Review by EiffelFly — APPROVED** [review]
@CaCaBlocker Everything are working. Thank for your contribution!!

**Pull request merged**